src/pages/journal/journal.reducer.ts

```diff
@@ -41,6 +41,25 @@ export function journalReducer(state = initialState, action: journalActions.Type
       return {
         ...stateWithoutError,
       }
+    case journalActions.CLEAR_CHANGED_SLOT:
+      const slots = state.slots[state.selectedDate].map( (slot) => {
+       if ( slot.slotData.slotDetail.slotId === action.slotId) {
```

Comment:
Could you please use Lodash `get` or `has` here to make this check resilient to undefined? For future slot types, e.g. `PersonalCommitment`, I don't think we will necessarily have the slot detail.

Reply:
Great catch. Have added lodash has checks on slotDetail existing and on slotId existing.